lib/streamlit/widgets.py

```diff
@@ -158,3 +158,25 @@ def _get_widget_value(state: Optional[WidgetState]) -> Optional[Any]:
         return json.loads(getattr(state, value_type))
 
     return getattr(state, value_type)
+
+
+def beta_widget_value(key: str) -> Any:
+    """Returns the value of a widget with a given id, for use in
+    state update callbacks.
+    """
+    import streamlit.report_thread as ReportThread
+    from streamlit.server.server import Server
+    import streamlit.elements.utils as utils
+
+    ctx = ReportThread.get_report_ctx()
+
+    if ctx is None:
+        return None
+
+    this_session = Server.get_current().get_session_by_id(ctx.session_id)
+    widget_states: WidgetStateManager = this_session.get_widget_states()
+
+    widget_id = utils._get_widget_id("", None, key)
+    deserializer = widget_states._widget_deserializers.get(widget_id, lambda x: x)
```

Comment:
I forgot something here. I realized one thing that some deserializers are not registerred, this is because we only register deserializers [if there is a callback](https://github.com/streamlit/streamlit/blob/session-state/lib/streamlit/elements/utils.py#L145)

I _think_ all that needs to be done is to add the deserializer in any case, which is hopefully a 1-3 line change.